test/splitgraph/commands/test_aggregation_pushdown.py

```diff
@@ -0,0 +1,392 @@
+    query = """WITH sub_agg AS (
+        SELECT state, gender, min(age), max(balance) as max_balance
+        FROM es.account GROUP BY state, gender
+    )
+    SELECT min(max_balance), gender FROM sub_agg GROUP BY gender"""
+
+    # Only the subquery is pushed-down, with no redundant aggregations
+    result = get_engine().run_sql("EXPLAIN " + query)
+    assert len(result) > 2
+    assert _extract_query_from_explain(result) == {
+        "aggs": {
+            "group_buckets": {
+                "composite": {
+                    "sources": [
+                        {"state": {"terms": {"field": "state"}}},
+                        {"gender": {"terms": {"field": "gender"}}},
+                    ],
+                    "size": 5,
+                },
+                "aggregations": {"max.balance": {"max": {"field": "balance"}}},
+            }
+        }
+    }
+
+    # Ensure results are correct
+    result = get_engine().run_sql(query)
+    assert len(result) == 2
+
+    # Assert aggregation result
+    assert result == [(37358, "M"), (31968, "F")]
+
+
+@pytest.mark.mounting
+def test_elasticsearch_not_pushed_down(local_engine_empty):
+    _mount_elasticsearch()
+
+    _bare_filtering_query = {"query": {"bool": {"must": [{"range": {"age": {"gt": 30}}}]}}}
+
+    # Aggregation only filtering is not going to be pushed down
+    result = get_engine().run_sql(
+        "EXPLAIN SELECT max(balance), avg(age) FROM es.account WHERE age > 30"
+    )
+    assert len(result) > 2
+    assert _extract_query_from_explain(result) == _bare_filtering_query
+
+    # Grouping only filtering is not going to be pushed down
+    result = get_engine().run_sql("EXPLAIN SELECT age FROM es.account WHERE age > 30 GROUP BY age")
+    assert len(result) > 2
+    assert _extract_query_from_explain(result) == _bare_filtering_query
+
+    # Grouping only post-aggregation filtering is not going to be pushed down either
+    result = get_engine().run_sql("EXPLAIN SELECT age FROM es.account GROUP BY age HAVING age > 30")
+    assert len(result) > 2
+    assert _extract_query_from_explain(result) == _bare_filtering_query
+
+    # Aggregation and grouping filtering is not going to be pushed down
+    result = get_engine().run_sql(
+        "EXPLAIN SELECT age, min(balance) FROM es.account WHERE age > 30 GROUP BY age"
+    )
+    assert len(result) > 2
+    assert _extract_query_from_explain(result) == _bare_filtering_query
+
+    _bare_sequential_scan = {"query": {"bool": {"must": []}}}
+
+    # SELECT STAR is not going to be pushed down
+    result = get_engine().run_sql("EXPLAIN SELECT * FROM es.account")
+    assert len(result) > 2
+    assert _extract_query_from_explain(result) == _bare_sequential_scan
+
+    # DISTINCT queries are not going to be pushed down
+    result = get_engine().run_sql("EXPLAIN SELECT COUNT(DISTINCT city) FROM es.account")
+    assert len(result) > 2
+    assert _extract_query_from_explain(result) == _bare_sequential_scan
```

Comment:
Idea for a test: some kind of a JOIN here, e.g. a JOIN between two aggregation results on `es.account` (I think it won't be pushed down, but useful to test anyway)

Reply:
Another idea: a nested expression, e.g. `SELECT avg(age * balance) FROM es.account GROUP BY state` to make sure this doesn't crash

Reply:
Yet another: a window query that I think might be implemented as a sort/groupby, e.g. the balance of the oldest person in each state (haven't tested that the syntax is correct):

```sql
SELECT
	age, balance, state
	RANK () OVER (
        PARTITION BY state
		ORDER BY age DESC
	) rank_number 
FROM
	es.accounts
WHERE rank_number = 1
```

Reply:
Thanks, good ideas!

I'll add them to the test suites.